### PR TITLE
Ensure DDL dialect queries propagate errors back to descendent scopes.

### PR DESCRIPTION
**Pull request by jaytaylor**
Ensure DDL dialect queries propagate error states to descendent scopes.

Includes relevant unit-test.

Branched from jay/current_database (please merge that branch first!).

### CODE REVIEW

**Comment by jaytaylor** [comment]
This is now ready for review.

File: main.go [hunk 1]
```diff
@@ -365,7 +365,10 @@ func (s *DB) Rollback() *DB {
 }
 
 func (s *DB) NewRecord(value interface{}) bool {
-	return s.clone().NewScope(value).PrimaryKeyZero()
+	scope := s.clone().NewScope(value)
+	result := scope.PrimaryKeyZero()
+	s.err(scope.db.Error)
```
Comment (jinzhu) [inline main.go]: This `s.err()`  is not necessary, there will be SQL be invoked here.

Comment (jaytaylor) [inline main.go]: Without it I don't think the unit-test will pass.  I will double check and provide more details when I get home.

**Comment by jaytaylor** [comment]
@jinzhu You were spot on - it works fine w/o those lines so I've reverted `NewRecord` back to the original logic.

Thanks!

**Pull request merged**